ops/instrument_pam.go

```diff
@@ -0,0 +1,153 @@
+package ops
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/unit"
+	"go.opentelemetry.io/otel/semconv/v1.17.0"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/valkyrie-fnd/valkyrie/internal"
+	"github.com/valkyrie-fnd/valkyrie/pam"
+	"github.com/valkyrie-fnd/valkyrie/pam/genericpam"
+	"github.com/valkyrie-fnd/valkyrie/pam/vplugin"
+)
+
+const (
+	VPluginName    = "pam-vplugin-client"
+	GenericPAMName = "genericpam-client"
+	RPCSystem      = "net/rpc"
+	RPCService     = "vplugin.PluginPAM"
+)
+
+var rpcAttributes = []attribute.KeyValue{semconv.RPCSystemKey.String(RPCSystem), semconv.RPCService(RPCService)}
+
+func init() {
+	vplugin.Pipeline.Register(pamTracingHandler(VPluginName, rpcAttributes...),
```

Comment:
I wonder if the pipeline registration should be done in a more central place. Or in the vplugin/genericpam packages themselves?

Reply:
Same reasoning here as for rest, didn't want vplugin/genericpam to depend on ops/otel etc. The goal is to encapsulate all otel imports into the ops package only and not have it leak into all other packages too